library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/WireframeMapper.kt

```diff
@@ -25,6 +26,12 @@ interface WireframeMapper<in T : View, out S : MobileSegment.Wireframe> {
      * system state.
      * @see MobileSegment.Wireframe
      * @see SystemInformation
+     * @param recordedDataQueueRefs in which we maintain references to the work queue for async
+     * updates.
      */
-    fun map(view: T, mappingContext: MappingContext): List<S>
+    fun map(
+        view: T,
+        mappingContext: MappingContext,
+        recordedDataQueueRefs: RecordedDataQueueRefs? = null
```

Comment:
why is this optional ? Is there any case where this could be null ?

Reply:
My consideration here was that there are places in the code where it isn't relevant to pass the refs, and we'd need to store them to support doing so - for example SwitchCompatMapper has a resolveMainWireframes that calls map in a subclass of CheckableWireframeMapper that actually receives the refs. On the other hand, having it nullable does seem error prone in the sense that it might not be passed somewhere it was needed and would be difficult to notice